src/test/java/programminglife/parser/GraphParserTest.java

```diff
@@ -122,4 +137,112 @@ public void update(Observable o, Object arg) {
             }
         }
     }
+
+    @Test
+    public void parseSegmentNotStartingS() {
+        try {
+            graphParser.parseSegment("Fiets");
+            fail("This should not be parsed");
+        } catch (Exception e) {
+            assertEquals(UnknownTypeException.class, e.getClass());
+            assertEquals("Line (Fiets) is not a segment", e.getMessage());
+        }
+    }
+
+    @Test
+    public void parseLinkNotStartingL() throws UnknownTypeException {
+        try {
+            graphParser.parseLink("Fiets");
+            fail("This should not be parsed");
+        } catch (Exception e) {
+            assertEquals(UnknownTypeException.class, e.getClass());
+            assertEquals("Line (Fiets) is not a link", e.getMessage());
+        }
+    }
+
+    @Test
+    public void parseHeaderNotStartingH() throws UnknownTypeException {
+        try {
+            graphParser.parseHeader("Fiets");
+            fail("This should not be parsed");
+        } catch (Exception e) {
+            assertEquals(UnknownTypeException.class, e.getClass());
+            assertEquals("Line (Fiets) is not a header", e.getMessage());
+        }
+    }
+
+    @Test
+    public void parseMultipleHeaders() throws Exception {
+        String[] lines = new String[] {"H\tVN:Z:1.0\n", "H\tBUILD:Z:VCF2GRAPH\n", "H\tORI:Z:SZAXPI008746-45"};
+        for (String line : lines) {
+            graphParser.parseHeader(line);
+        }
+        assertTrue(graphParser.getGraph().getGenomeNames().contains("SZAXPI008746-45"));
+    }
+
+    @Test
+    public void parseSegmentIDNotNumber() throws UnknownTypeException {
+        try {
+            graphParser.parseSegment("S\tID\tT\t*\tORI:Z:TKK-01-0066.fasta;TKK_REF.fasta");
+            fail("This should not be parsed");
+        } catch (Exception e) {
+            assertEquals(UnknownTypeException.class, e.getClass());
+            assertEquals("The segment ID (ID) should be a number", e.getMessage());
+        }
```

Comment:
These should use `@Test(expected = UnnownTypeException.class)` as that is more robust and removes anyneeds to use `fail()`